hydra-main/src/main/java/com/addthis/hydra/job/store/JdbcDataStore.java

```diff
@@ -0,0 +1,430 @@
+            if (!hasData) {
+                return null;
+            } else {
+                firstValue = dbTypeToValue(resultSet.getObject(1, getValueType()));
+            }
+            boolean moreResults = resultSet.next();
+            // Given the UNIQUE constraint, it would be extremely unexpected to find multiple values for a single path/childId
+            if (moreResults) {
+                throw new RuntimeException("Found multiple results after expecting a unique result; bailing");
+            }
+            return firstValue;
+        }
+    }
+
+    private void delete(String path, String childId) throws SQLException {
+        try (Connection connection = cpds.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(getDeleteTemplate());
+            preparedStatement.setString(1, path);
+            preparedStatement.setString(2, childId);
+            executeAndTimeInsert(preparedStatement);
+        }
+    }
+
+    @Override
+    public void putAsChild(String parent, String childId, String value) throws Exception {
+        checkValidKeys(parent, childId);
+        insert(parent, childId, value);
+    }
+
+    @Override
+    public String getChild(String parent, String childId) throws Exception {
+        checkValidKeys(parent, childId);
+        return querySingleResult(parent, childId);
+    }
+
+    @Override
+    public void deleteChild(String parent, String childId) {
+        checkValidKeys(parent, childId);
+        try {
+            delete(parent, childId);
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void delete(String path) {
+        try {
+            checkValidKey(path);
+            delete(path, blankChildValue);
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    /**
+     * Find the names of all children of a given path.
+     */
+    public List<String> getChildrenNames(String path) {
+        checkValidKey(path);
+        try (Connection connection = cpds.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(getGetChildNamesTemplate());
+            preparedStatement.setString(1, path);
+            preparedStatement.setString(2, blankChildValue);
+            ResultSet resultSet = executeAndTimeQuery(preparedStatement);
+            boolean hasResults = resultSet.next();
+            List<String> rv = new ArrayList<>();
+            if (!hasResults) {
+                return rv;
+            }
+            do {
+                String key = resultSet.getString(1);
+                if (key != null) {
+                    rv.add(key);
+                }
+            } while (resultSet.next());
+            return rv;
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
```

Comment:
our code formatting guidelines prefer not to put javadoc inbetween annotations and methods